.github/workflows/build.yml

```diff
@@ -0,0 +1,72 @@
+name: build and test
+on:
+  push:
+    branches-ignore:
+      - "gh-pages"
+    tags-ignore:
+      - "v0.[0-9]**"
+  pull_request:
+    branches-ignore:
+      - "gh-pages"
+
+jobs:
+  build_and_test:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+
+          # GCC builds
+          - CC: gcc
+            CONF_FLAGS:
+            CHECK: 1
+            DISTCHECK: 1
+            name_extra: Basic build and dist check
+
+          - CC: gcc
+            CONF_FLAGS: --without-simd
+            name_extra: Without simd
+
+          - CC: gcc
+            CONF_FLAGS: --host=i686-linux
+            CFLAGS: -m32
+            arch: i386
+            name_extra: 32-bit build
+
+          # clang builds
+          - CC: clang
+            CONF_FLAGS:
+            CHECK: 1
+            DISTCHECK: 1
+            name_extra: Basic build and dist check
+
+          - CC: clang
+            CONF_FLAGS: --without-simd
+            name_extra: Without simd
+
+          - CC: clang
+            CONF_FLAGS: --host=i686-linux
+            CFLAGS: -m32
+            arch: i386
+            name_extra: 32-bit build
+
+    name: ${{ matrix.name_extra }} (${{ matrix.CC }})
+    runs-on: ubuntu-latest
+    env:
+      CC: ${{ matrix.CC }}
+      CFLAGS: ${{matrix.CFLAGS }}
+      XRDP_CFLAGS: -I${{ github.workspace }}/xrdp/common
+
+    steps:
+      - uses: actions/checkout@v2
+      - run: sudo scripts/install_xorgxrdp_build_dependencies_with_apt.sh ${{ matrix.arch }} --allow-downgrades --allow-remove-essential --allow-change-held-packages
+      - run: git clone --depth 1 --branch=devel https://github.com/neutrinolabs/xrdp.git ${{ github.workspace}}/xrdp
+      - run: ./bootstrap
+      - run: ./configure ${{ matrix.CONF_FLAGS }}
+      - run: make CFLAGS="$CFLAGS -O2 -Wall -Wwrite-strings -Werror"
+      - if: ${{ matrix.CHECK }}
+        run: make check
+      - if: ${{ matrix.DISTCHECK }}
+        run: make distcheck
+      - name: Display test logs (make check only)
+        run: scripts/display_test_logs
```

Comment:
I think `if: failed()` or `if: always()` is required to make sure the logs get output when there is a failure in a previous step.

Reply:
Quite right - I verified this on a temp branch